python/ray/autoscaler/_private/vsphere/pyvmomi_sdk_provider.py

```diff
@@ -25,7 +25,37 @@ def __init__(self, server, user, password, session_type: Constants.SessionType):
         atexit.register(Disconnect, smart_connect_obj)
         self.pyvmomi_sdk_client = smart_connect_obj.content
 
-    def get_pyvmomi_obj(self, vimtype, name):
+    def get_pyvmomi_obj_by_moid(self, vimtype, moid):
+        """
+        This function finds the vSphere object by the object moid and the object type.
+        The object type can be "VM", "Host", "Datastore", etc.
+        The object moid is a unique id for this object type under the vCenter server.
+        To check all such object information, you can go to the managed object board
+        page of your vCenter Server, such as: https://<your_vc_ip/mob
+        """
+        obj = None
+        if self.pyvmomi_sdk_client is None:
+            raise ValueError("Must init pyvmomi_sdk_client first.")
+
+        container = self.pyvmomi_sdk_client.viewManager.CreateContainerView(
+            self.pyvmomi_sdk_client.rootFolder, vimtype, True
+        )
+
+        for c in container.view:
+            if moid:
+                if moid in str(c):
+                    obj = c
+                    break
+            else:
+                obj = c
+                break
+        if not obj:
+            raise RuntimeError(
+                f"Unexpected: cannot find vSphere object {vimtype} with moid: {moid}"
+            )
+        return obj
+
+    def get_pyvmomi_obj_by_name(self, vimtype, name):
```

Comment:
It looks like these two functions are almost identical. Can we refactor it to avoid duplicated code?

Reply:
I don't think we can do a big imporvement on that, what we can do is to add another parameter to indicate that this is a name or a ID and leave on function. But to achieve this we need to involve anther enum type. I would suggest we don't change this, because there will not be function like this again.